database/migrations/2024_04_04_000001_add_eligible_main_rulesets_to_beatmapsets.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::table('osu_beatmapsets', function (Blueprint $table) {
+            $table->json('eligible_main_rulesets')->nullable();
```

Comment:
I feel like this should be mode instead of ruleset and all the ruleset stuff in beatmap discussions/nominations reverted since all the existing ones are already stored using the legacy name, and the switch to `Ruleset` makes all the logic more confusing.

Reply:
more like should store ruleset id instead, I think

Reply:
switched to storing the ids instead, not sure if better but at least they're in a sorted now 👀